### PR TITLE
Corey Sheesley

### PR DESCRIPTION
**Pull request by CSheesley**

### CODE REVIEW

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 1]
```diff
@@ -105,6 +109,10 @@
 
     # ------------------ Using ActiveRecord ----------------------
     # Solution goes here
+    # orders_of_500_and_700 = Order.where(amount: 500).or(Order.where(amount: 700))
+    # orders_of_700_and_1000 = Order.where(amount: 700).or(Order.where(amount: 1000))
+    orders_of_500_and_700 = Order.where("amount = ? OR amount = ?", 500, 700)
+    orders_of_500_and_700 = Order.where("amount = ? OR amount = ?", 700, 1000)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: shorter: `Order.where(amount: [500,700])`

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 2]
```diff
@@ -136,7 +145,7 @@
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    orders_2 = Order.find(ids_to_find).sort
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: TODO: how can you do this without a Ruby sort on the end?

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 3]
```diff
@@ -150,7 +159,8 @@
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    #is this correct? .where returns relations, not an array, but test passes.
+    orders_between_700_and_1000  = Order.where(:amount => 700 ..1000)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: the 'hashrocket' syntax isn't really used much any more and `(amount: 700..1000)` hash syntax is preferable instead.

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 4]
```diff
@@ -165,7 +175,7 @@
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    orders_less_than_550  = Order.where("amount < 550")
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: don't forget to use binding parameters where you can, like `('amount < ?', 550)`

File: spec/models/activerecord_obstacle_course_spec.rb (hunk 2 above)
Comment (CSheesley) [reply spec/models/activerecord_obstacle_course_spec.rb]: orders = Order.order(:id).find(ids_to_find)

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 5]
```diff
@@ -141,11 +141,11 @@
     ids_to_find = [@order_1.id, @order_3.id, @order_5.id, @order_7.id]
 
     # ----------------------- Using Ruby -------------------------
-    orders = Order.all.select { |order| ids_to_find.include?(order.id) }
+    # orders = Order.all.select { |order| ids_to_find.include?(order.id) }
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    orders_2 = Order.find(ids_to_find).sort
+    orders = Order.order(:id).find(ids_to_find)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: better: `Order.where(id: ids_to_find)`

**Pull request closed without merging**